### PR TITLE
[MTF-593] Remove LKV check from Atomic Cache Gem.

### PR DESCRIPTION
**Pull request by AmishNinja**
* Remove LKV check from wait loop of atomic cache.
* Increment patch version.
* Remove LKV test from atomic_cache_client_spec.rb.

Tasks
-----
* [x] [Code of Conduct](https://github.com/Ibotta/atomic_cache/blob/main/CODE_OF_CONDUCT.md) reviewed
* [x] Specs written and passing
* [x] Increment the version file (`./lib/atomic_cache/version.rb`) when applicable
    * See [semver.org](https://semver.org/) for what segment to increment

### CODE REVIEW

File: lib/atomic_cache/version.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 module AtomicCache
-  VERSION = "0.5.2.rc1"
+  VERSION = "0.5.3.rc1"
```
Comment (lebeerman) [inline lib/atomic_cache/version.rb:4]: We chatted over this and Im unclear - maybe a `0.6` version is needed? We arent observing the `wait.klv.present` [increment in the dashboard](https://app.datadoghq.com/dashboard/s58-faw-q7b/memcached-atomic-cache?from_ts=1627676917563&to_ts=1628281717563&live=true)...

Comment (AmishNinja) [reply lib/atomic_cache/version.rb:4]: I talked with Titus about this, and he says that a patch version makes the most sense, as while it is not a bug fix per se, it is not a very noticeable change.

**Review by lebeerman — APPROVED** [review]
LGTM. Only request would be to add a link/title to the pr for search-ability/archaeology

**Comment by AmishNinja** [comment]
> LGTM. Only request would be to add a link/title to the pr for search-ability/archaeology

Good call; added one in. Let me know if the wording should be different!

**Pull request merged**